clap_blocks/src/compactor.rs

```diff
@@ -1,181 +1,220 @@
-
-    /// Min number of recent ingested files a partition needs to be considered for compacting
-    /// Default: 1
-    #[clap(
-        long = "--compaction-min-number-recent-ingested-files-per-partition",
-        env = "INFLUXDB_IOX_COMPACTION_MIN_NUMBER_RECENT_INGESTED_FILES_PER_PARTITION",
-        default_value = "1",
-        action
-    )]
-    pub min_number_recent_ingested_files_per_partition: usize,
-
-    /// A compaction operation for hot partitions will gather as many L0 files with their
-    /// overlapping L1 files to compact together until the total size of input files crosses this
-    /// threshold. Later compactions will pick up the remaining L0 files.
-    ///
-    /// A compaction operation will be limited by this or by the file count threshold, whichever is
-    /// hit first.
-    ///
-    /// Default is 1024 * 1024 * 100 = 100,048,576 bytes (100MB).
-    #[clap(
-        long = "--compaction-input-size-threshold-bytes",
-        env = "INFLUXDB_IOX_COMPACTION_INPUT_SIZE_THRESHOLD_BYTES",
-        default_value = "100048576",
-        action
-    )]
-    pub input_size_threshold_bytes: u64,
-
-    /// A compaction operation for cold partitions will gather as many L0 files with their
-    /// overlapping L1 files to compact together until the total size of input files crosses this
-    /// threshold. Later compactions will pick up the remaining L0 files.
-    ///
-    /// Default is 1024 * 1024 * 600 = 629,145,600 bytes (600MB).
-    #[clap(
-        long = "--compaction-cold-input-size-threshold-bytes",
-        env = "INFLUXDB_IOX_COMPACTION_COLD_INPUT_SIZE_THRESHOLD_BYTES",
-        default_value = "629145600",
-        action
-    )]
-    pub cold_input_size_threshold_bytes: u64,
-
-    /// A compaction operation will gather as many L0 files with their overlapping L1 files to
-    /// compact together until the total number of L0 + L1 files crosses this threshold. Later
-    /// compactions will pick up the remaining L0 files.
-    ///
-    /// A compaction operation will be limited by this or by the input size threshold, whichever is
-    /// hit first.
-    ///
-    /// Default is 50.
-    #[clap(
-        long = "--compaction-input-file-count-threshold",
-        env = "INFLUXDB_IOX_COMPACTION_INPUT_FILE_COUNT_THRESHOLD",
-        default_value = "50",
-        action
-    )]
-    pub input_file_count_threshold: usize,
-
-    /// The multiple of times that compacting hot partitions should run for every one time that
-    /// compacting cold partitions runs. Set to 1 to compact hot partitions and cold partitions
-    /// equally.
-    ///
-    /// Default is 4.
-    #[clap(
-        long = "--compaction-hot-multiple",
-        env = "INFLUXDB_IOX_COMPACTION_HOT_MULTIPLE",
-        default_value = "4",
-        action
-    )]
-    pub hot_multiple: usize,
+#![cfg_attr(rustfmt, rustfmt_skip)] // https://github.com/rust-lang/rustfmt/issues/5489
+
+macro_rules! gen_compactor_config {
```

Comment:
Some comments here might help, especially with the rationale for this macro

For example:

```suggestion
/// Create compactor configuration that can have different defaults
macro_rules! gen_compactor_config {

```